bin/oneoff/data_fix/delete_projects_for_purged_users

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+require 'cdo/delete_accounts_helper'
+
+# Delete Project S3 Objects created by Users whose accounts were purged if Project data is restored from offsite backup.
+# Student project S3 Objects are irrevocably deleted by the Account Purger (28 days after the user's account is
+# soft deleted). But, S3 Bucket Replication (used to backup project buckets to offsite account for disaster recovery)
+# intentionally does not propagate Object deletion to the backup to prevent malicious deletions from propagating to the
+# backup. In the unlikely recovery scenario that the projects system is restored from the offsite backup, use this script
+# to delete S3 Objects for projects belonging to purged users.
+
+options = {actually_delete: false}
+OptionParser.new do |opts|
+  opts.banner = "Usage: #{File.basename(__FILE__)} [options]"
+  opts.on('-d', '--actually-delete', 'Actually perform the delete.') do
+    options[:actually_delete] = true
+  end
+  opts.on('-h', '--help', 'Add -d to perform the delete.') do
+    puts opts
+    exit
+  end
+end.parse!
+CDO.log.info "Called with options: #{options}"
+
+start_user_id = User.first.id
+end_user_id = User.last.id
+
+BATCH_SIZE = 200
+
+users_purged_of_project_objects = 0
+users_with_project_purge_errors = 0
+User.
+  with_deleted.
+  where.not(purged_at: nil).
+  find_in_batches(start: start_user_id, finish: end_user_id, batch_size: BATCH_SIZE) do |user_batch|
+  user_batch.each do |user|
+    if options[:actually_delete]
+      delete_project_backed_progress(user)
+      users_purged_of_project_objects += 1
+      CDO.log.info "Deleting project S3 Objects for purged user: id=#{user.id}"
+    end
+  rescue => error
+    users_with_project_purge_errors += 1
+    CDO.log.error error
```

Comment:
This one-off cleanup script would be run manually by an engineer, so they would review the output and decide what to do about the (hopefully) handful of purged users where there was an error re-deleting their project objects.

The cron job that detects that the project system has been restored [Raises an Error](https://github.com/code-dot-org/code-dot-org/blob/ba1afd4967b4368b0adfaa59e8676ba1bdd3c351/bin/cron/detect_projects_restored_from_backup#L38), which our cron job wrapper detects and uses to create a HoneyBadger Error. I think I need to improve the information in the error that is raised so that the Developer of the Day knows that we need to manually run the clean up script